### ISSUE TITLE
migration:generate recreates defaults for no reason each migration

### ISSUE THREAD

**Issue by Satont** [typeorm/typeorm#9750]
### Issue description

When i generates new migrations, then typeorm always trying to recreate some defualts

### Expected Behavior

Typeorm shouldn't try to recreate defauls

### Actual Behavior

Typeorm always recreates defaults.

### Steps to reproduce

This is what typeorm generates
```js
await queryRunner.query(`ALTER TABLE "channel_events_donations" ADD "donateId" character varying`);
        await queryRunner.query(`ALTER TABLE "channels_modules_settings" ALTER COLUMN "id" DROP DEFAULT`);
        await queryRunner.query(`ALTER TABLE "channels_modules_settings" ALTER COLUMN "id" SET DEFAULT gen_random_uuid()`);
        await queryRunner.query(`ALTER TABLE "channels_requested_songs" ALTER COLUMN "id" DROP DEFAULT`);
        await queryRunner.query(`ALTER TABLE "channels_requested_songs" ALTER COLUMN "id" SET DEFAULT gen_random_uuid()`);
        ```

My entity
```
@Entity('channels_modules_settings')
export class ChannelModuleSettings {
  @PrimaryColumn('uuid', {
    primary: true,
    name: 'id',
    default: () => 'gen_random_uuid()',
  })
  id: string;

  @Column('enum', { enum: ModuleType })
  type: ModuleType;

  @Column('jsonb')
  settings: Record<string, any>;

  @Column()
  channelId: string;

  @ManyToOne('Channel', 'modules')
  @JoinColumn({ name: 'channelId' })
  channel?: Relation<Channel>;
}
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| Node.js version     | 18.2.1  |
| Typescript version  | 4.9 |
| TypeORM version     | 0.3.10  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, but I don't know how to start. I would need guidance.

**Comment by jleverenz** [comment]
I just ran into a similar issue, due to the `default: () => 'gen_random_uuid()'` configuration. Re-reading the TypeORM docs, switching to one of the `Generated` decorators solved it for me, it might work for you. Try:

```
@Entity('channels_modules_settings')
export class ChannelModuleSettings {
  @PrimaryGeneratedColumn('uuid')
  id: string;
  
  // ...
}
```

I *think* you might get the same effective sql generated, without the repeated migrations.

**Comment by kacsablabla** [comment]
I've had the same issue when upgrading 0.3.11 to 0.3.16. I reverted it, and works again.

**Comment by N-Andronopoulos** [comment]
I have the same issue with mysql2 v2 or v3 driver after version 0.3.10 (tried 0.3.12 and 0.3.16 issue exists in both). 
It tries to recreate ids and relations FKs etc.
e.g.
```  
await queryRunner.query(`ALTER TABLE \`waiter_ring_actions\` DROP PRIMARY KEY`);
await queryRunner.query(`ALTER TABLE \`waiter_ring_actions\` DROP COLUMN \`id\``);
await queryRunner.query(`ALTER TABLE \`waiter_ring_actions\` ADD \`id\` varchar(36) NOT NULL PRIMARY KEY`);
        await queryRunner.query(`ALTER TABLE \`order_items\` ADD CONSTRAINT \`FK_22866dffffb1fc32a384ecf3b0c\` FOREIGN KEY (\`cartId\`) REFERENCES \`orders\`(\`id\`) ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE \`order_history\` ADD CONSTRAINT \`FK_464d9910b78618abc044b7aec7f\` FOREIGN KEY (\`cartId\`) REFERENCES \`orders\`(\`id\`) ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE \`order_history_items\` ADD CONSTRAINT \`FK_e7c502e4d28c16d4247fb5c231e\` FOREIGN KEY (\`itemId\`) REFERENCES \`order_history\`(\`id\`) ON DELETE NO ACTION ON UPDATE NO ACTION`);
```

**Comment by arminhupka** [comment]
> I have the same issue with mysql2 v2 or v3 driver after version 0.3.10 (tried 0.3.12 and 0.3.16 issue exists in both). It tries to recreate ids and relations FKs etc.
> 
> We use `@PrimaryGeneratedColumn('uuid')` on all entities as well.
> 
> e.g.
> 
> ```
> await queryRunner.query(`ALTER TABLE \`ring_actions\` DROP PRIMARY KEY`);
> await queryRunner.query(`ALTER TABLE \`ring_actions\` DROP COLUMN \`id\``);
> await queryRunner.query(`ALTER TABLE \`ring_actions\` ADD \`id\` varchar(36) NOT NULL PRIMARY KEY`);
> await queryRunner.query(`ALTER TABLE \`order_items\` ADD CONSTRAINT \`FK_22866dffffb1fc32a384ecf3b0c\` FOREIGN KEY (\`cartId\`) REFERENCES \`orders\`(\`id\`) ON DELETE NO ACTION ON UPDATE NO ACTION`);
> await queryRunner.query(`ALTER TABLE \`order_history\` ADD CONSTRAINT \`FK_464d9910b78618abc044b7aec7f\` FOREIGN KEY (\`cartId\`) REFERENCES \`orders\`(\`id\`) ON DELETE NO ACTION ON UPDATE NO ACTION`);
> await queryRunner.query(`ALTER TABLE \`order_history_items\` ADD CONSTRAINT \`FK_e7c502e4d28c16d4247fb5c231e\` FOREIGN KEY (\`itemId\`) REFERENCES \`order_history\`(\`id\`) ON DELETE NO ACTION ON UPDATE NO ACTION`);
> ```

I had the same issue and I decided to update to @next version and everything is now is ok

**Comment by ankitchhatbar** [comment]
The below works well for me
```JavaScript
@Generated("uuid")
@PrimaryColumn({ length: 36 })
id: string
```

**Comment by N-Andronopoulos** [comment]
Linked to #10090

**Comment by pleerock** [comment]
Must be fixed by #10041

**Comment by sirmonin** [comment]
@pleerock this issue has been fixed for MySQL only. PostgreSQL is mentioned in the issue description by @Satont, and I can confirm that even 0.3.20 version still has this issue. Please, reopen this.
